lib/rspec/core/output_wrapper.rb

```diff
@@ -0,0 +1,26 @@
+module RSpec
+  module Core
+    # @private
+    class OutputWrapper
+      # @private
+      attr_writer :output
+
+      # @private
+      def initialize(output)
+        @output = output
+      end
+
+      # Redirect calls for IO interface methods
+      (IO.instance_methods - Enumerable.instance_methods - Object.instance_methods + [:to_str]).each do |method|
+        define_method(method) do |*args, &block|
+          @output.send(method, *args, &block)
+        end
+      end
+
+      # @private
+      def respond_to?(*args)
+        args[0] == :output= || @output.respond_to?(*args)
+      end
```

Comment:
Given this class is designed to wrap an `IO` object, and a string is _not_ an `IO` object, I think we should not wrap a string with this class (and therefore, you shouldn't need this extra logic).  Instead, can the wrapping be delayed until the file identified by the string is opened?